streaming/src/test/scala/org/apache/spark/streaming/scheduler/ReceiverTrackerSuite.scala

```diff
@@ -110,8 +110,15 @@ class ReceiverTrackerSuite extends TestSuiteBase {
   }
 }
 
-/** An input DStream with a hard-coded receiver that gives access to internals for testing. */
-private class RateLimitInputDStream(@transient ssc_ : StreamingContext)
+/**
+ * An input DStream with a hard-coded receiver that gives access to internals for testing.
+ *
+ * @note Make sure to call {{{SingletonDummyReceiver.reset()}}} before using this in a test,
+ *       or otherwise you may get {{{NotSerializableException}}} when trying to serialize
+ *       the receiver.
```

Comment:
Because after being `started` once, it has a reference to the ReceiverSupervisor, which is not serializable. So the second time the receiver is used, it can't be serialized and shipped to executors. I thought the one-line method is self-explanatory, but I can add a few more lines of comments around that one line :)